pykafka/simpleconsumer.py

```diff
@@ -206,6 +206,7 @@ def _update(self):
         with self._update_lock:
             self._cluster.update()
             self._setup_partitions_by_leader()
+            self._discover_offset_manager()
```

Comment:
In some cases that will happen, but in the one I'm testing now, [this case](https://github.com/Parsely/pykafka/blob/4ea463f18f74b38b1da72507ca54f6b0316c5afc/pykafka/cluster.py#L247) is hit instead. I think it's correct, it just makes this problem a lot more gnarly.